### ISSUE TITLE
Error codes returned by characteristic setters are not forwarded to HomeKit

### ISSUE THREAD

**Issue by rutzfutz** [homebridge/HAP-NodeJS#867]
**Describe The Bug:**

In HAP-NodeJS `0.9.0-beta.97`, if one of my setters returns a `HAPStatusError`, either through the first callback argument, or by throwing it in the `.onSet()` function, the given error code never reaches HomeKit.

Instead, the error code sent to HomeKit is always `HAPStatus.SERVICE_COMMUNICATION_FAILURE`.

**To Reproduce:**

```node
const {
	Accessory,
	Characteristic,
	Service,
	uuid,
	HapStatusError,
	HAPStatus
} = require('hap-nodejs')

var accessoryUUID = uuid.generate('hap-nodejs:accessories:Light:testLight')

var accessory = exports.accessory = new Accessory('Test light', accessoryUUID)

const lightbulb = accessory.addService(new Service.Lightbulb())

lightbulb.getCharacteristic(Characteristic.On)
	.onSet((value) => {
		throw new HapStatusError(HAPStatus.INSUFFICIENT_AUTHORIZATION)
	})
	.onGet(() => {
		return false
	})
```

**Expected behavior:**

When attempting to toggle the test light on/off, it cannot be toggled. When opening the details, it shows that additional setup is required in the manufacturer's app.

**Actual behavior:**

When opening the details, it shows "no response".

**Comment by rutzfutz** [comment]
I created a pull request to fix this bug. I wasn't familiar enough with the tooling to create a meaningful test case to prevent future regressions, so I skipped this part. Hope it helps anyway.

**Comment by Supereg** [comment]
fixed by #866
